__tests__/ChannelPreviewCompact-test.js

```diff
@@ -0,0 +1,116 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+import React from 'react';
+import { cleanup, render, fireEvent } from '@testing-library/react';
+import { ChannelPreviewCompact } from '../src/components/ChannelPreviewCompact';
+import uuidv4 from 'uuid/v4';
+import { getTestClient, createUserToken } from './utils';
+
+// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
+// unmount and cleanup DOM after the test is finished.
+// eslint-disable-next-line no-undef
+afterEach(cleanup);
+const unreadCountClassName = 'str-chat__channel-preview-compact--unread';
+const activeChannelClassName = 'str-chat__channel-preview-compact--active';
+describe('ChannelPreviewCompact', () => {
+  let chatClient;
+  let userId;
+  let userToken;
+  let channelID;
+  let channel;
+  let channelName;
+  beforeAll(async function() {
+    chatClient = getTestClient();
+    userId = `thierry-${uuidv4()}`;
+    userToken = createUserToken(userId);
+    channelID = uuidv4();
+    chatClient.setUser(
+      {
+        id: userId,
+      },
+      userToken,
+    );
+    channelName = uuidv4();
+    channel = chatClient.channel('messaging', channelID, {
+      name: channelName,
+    });
+    await channel.create();
+    await channel.watch();
+  });
+
+  it('should call setActiveChannel on click', () => {
+    const setActiveChannel = jest.fn();
+    const watchers = {};
+    const { container } = render(
+      <ChannelPreviewCompact
+        latestMessage="This is a message from me, this is so cool man!!"
+        latestMessageLength={20}
+        channel={channel}
+        activeChannel={channel}
+        setActiveChannel={setActiveChannel}
+        watchers={{}}
+      />,
+    );
+
+    const containerBtn = container.querySelector(
+      '.str-chat__channel-preview-compact',
+    );
+    fireEvent.click(containerBtn);
+    expect(setActiveChannel).toHaveBeenCalledTimes(1);
+    expect(setActiveChannel).toHaveBeenLastCalledWith(channel, watchers);
+  });
+
+  it('should have str-chat__channel-preview-compact--active class only when channel is active', () => {
```

Comment:
tests based on css class is not exactly idea, but here thats what end user will see (different styles based on props), so making an exception here.